extensions/barrage/src/main/java/io/deephaven/extensions/barrage/util/BarrageStreamReader.java

```diff
@@ -259,7 +326,7 @@ public BarrageMessage safelyParseFrom(final StreamReaderOptions options,
             }
 
             deserializeTmConsumer.accept(System.nanoTime() - startDeserTm);
-            if (numAddBatchesRemaining + numModBatchesRemaining == 0) {
+            if (numAddRowsRead == numAddRowsTotal && numModRowsRead == numModRowsTotal) {
```

Comment:
We should also deliver this if `numAddRowsTotal == 0 && numModRowsTotal == 0` - maybe it would be worthwhile creating a `final boolean noDeephavenMetadata` earlier in the logic (to re-use above and here).